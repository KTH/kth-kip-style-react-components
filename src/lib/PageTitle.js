import React, { Component } from 'react'

class PageTitle extends Component {

  render () {
      /**
       * <PageTitle id='mainHeading' pageTitle="Redigera introduktion till kursen">
       *    <span>SF1624 Algebra och geometri 7,5 hp</span>
       * </PageTitle>
       */
    const { pageTitle, children, id } = this.props
    return (
      <div id={id} className={'pageTitle col ' + this.props.className}>
        <h1 data-testid='main-heading'>{pageTitle}</h1>
        <div data-testid='sub-heading'>
            {children}
        </div>
      </div>
    )
  }
}

export default PageTitle
