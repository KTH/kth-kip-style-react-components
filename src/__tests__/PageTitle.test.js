import React from 'react'
import renderer from 'react-test-renderer'
import PageTitle from '../lib/PageTitle'

const TEST_DATA = {
  mainTitle: 'Redigera introduktion till kursen',
  pageSubtitle: 'SF1624 Algebra och geometri 7,5 hp'
}

describe('PageTitle', () => {
  it('renders properly', () => {
    const tree = renderer
      .create(<PageTitle id='mainHeading' className='course-title' pageTitle={TEST_DATA.mainTitle}>
        <span>{TEST_DATA.pageSubtitle}</span>
      </PageTitle>
      ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
