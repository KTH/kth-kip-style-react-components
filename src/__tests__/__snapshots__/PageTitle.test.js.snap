// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle renders properly 1`] = `
<div
  className="pageTitle col course-title"
  id="mainHeading"
>
  <h1
    data-testid="main-heading"
  >
    Redigera introduktion till kursen
  </h1>
  <div
    data-testid="sub-heading"
  >
    <span>
      SF1624 Algebra och geometri 7,5 hp
    </span>
  </div>
</div>
`;
